File: queue/queue.py
import sys
sys.path.append('../doubly_linked_list')
from doubly_linked_list import DoublyLinkedList

"""
A queue is a data structure whose primary purpose is to store and
return elements in First In First Out order.
"""
class Queue:
    def __init__(self):
        self.size = 0
        self.storage = DoublyLinkedList()

    def __len__(self):
        return self.size

    def enqueue(self, item):
        self.storage.add_to_tail(item)
        self.size += 1

    def dequeue(self):
        if self.size == 0:
            return None
        else:
            self.size -=1
            return self.storage.remove_from_head()

File: data_structures/queue/queue.ts
import { DoublyLinkedList } from "../doubly_linked_list/doublyLinkedList";

export class Queue {
  constructor(
    public storage = new DoublyLinkedList(),
    public length: number = 0
  ) {}

  enqueue(item) {
    this.storage.addToTail(item);
    this.length++;
  }

  dequeue() {
    if (this.length === 0) {
      return null;
    } else {
      this.length--;
      return this.storage.removeFromHead();
    }
  }
}
